Desctiption: enable tetgen on systems, where it is available. Needs package-rebuilding
Author: Christophe Trophime <christophe.trophime@lncmi.cnrs.fr>
Reviewed-by: Anton Gladky <gladky.anton@gmail.com>
Last-Update: 2012-01-05

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -567,6 +567,18 @@
     message("WARNING: By including Tetgen you have to comply with Tetgen's "
             "special licensing requirements stated in contrib/Tetgen/LICENSE.")
   endif(ENABLE_TETGEN_NEW AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/contrib/TetgenNew/tetgen.h)
+  if(ENABLE_TETGEN)
+    find_library(TETGEN_LIB tet PATH_SUFFIXES lib)
+    if(TETGEN_LIB)
+      find_path(TETGEN_INC "tetgen.h" PATH_SUFFIXES src include tetgen)
+      if(TETGEN_INC)
+        set_config_option(HAVE_TETGEN "Tetgen")
+        add_definitions(-DTETLIBRARY)
+        list(APPEND EXTERNAL_LIBRARIES ${TETGEN_LIB})
+        list(APPEND EXTERNAL_INCLUDES ${TETGEN_INC})
+      endif(TETGEN_INC)
+    endif(TETGEN_LIB)
+  endif(ENABLE_TETGEN)
 endif(HAVE_MESH)
 
 if(ENABLE_FOURIER_MODEL)
